src/compiler/utilities.ts

```diff
@@ -1199,14 +1199,15 @@ export function getTokenPosOfNode(node: Node, sourceFile?: SourceFileLike, inclu
     // trivia for the list, we may have skipped the JSDocComment as well. So we should process its
     // first child to determine the actual position of its first token.
     if (node.kind === SyntaxKind.SyntaxList) {
-        const first = firstOrUndefined(getNodeChildren(node));
+        sourceFile ??= getSourceFileOfNode(node);

```

Comment:
Isn't this wasted work when the `getNodeChildren` function special cases `SyntaxKind.SyntaxList` to not use `sourceFile`?

Reply:
Apparently not - this is a recursive function, and at almost every other terminal branch the source file is consulted for something. So I can remove the check, but in practice we'll pull on `getSourceFileOfNode` at some point lower in the tree.

Happy to remove it if you all think it's cleaner to just pass `children` along.